app/javascript/components/widgets/twitter/tweet_body.jsx

```diff
@@ -3,40 +3,79 @@ import PropTypes from 'prop-types';
 import styled from 'styled-components';
 import Highlighter from 'react-highlight-words';
 import { fontSizes, colors } from '../../../lib/theme';
+import TweetContent from './tweet_content';
 
 const TweetText = styled.div`
   color: #959292;
   font-size: ${fontSizes.large};
   margin-bottom: 29px;
 `;
-const getStatusLink = text => {
+
+const getShortLinks = text => {
   const allWords = text.split(/[\s,\n]+/);
-  return allWords.filter(word => word.includes('https://t.co/'))[0];
+  return allWords.filter(word => word.includes('https://t.co/'));
 };
 
 const getMentionsAndTags = text => {
   const allWords = text.split(/[\s,\n]+/);
   return allWords.filter(word => word.includes('@') || word.includes('#'));
 };
 
-const TweetBody = ({ text }) => {
-  const displayText = text.replace(getStatusLink(text), ' ');
+const removeLinks = (text, linkedTweet) => {
+  const shortLinks = getShortLinks(text);
+  if (linkedTweet) return shortLinks.forEach(link => text.replace(link, ' '));
+  return text.replace(shortLinks[0], ' ');
+};
+
+const TweetBody = ({ text, mediaUrl, linkUrl }) => {
+  const displayText = removeLinks(text, linkUrl !== null);
+  const noContent = mediaUrl === null && linkUrl === null;
   return (
-    <TweetText>
-      <Highlighter
-        highlightStyle={{
-          backgroundColor: 'transparent',
-          color: `${colors.white}`,
-        }}
-        searchWords={getMentionsAndTags(displayText)}
-        textToHighlight={displayText}
-      />
-    </TweetText>
+    <React.Fragment>
+      <div>
+        {/* just content, no text */}
+        {typeof displayText === 'undefined' ? null : (
```

Comment:
I'm confused about the comment 
> {/* just content, no text */}

Isn't the below code the Twitter text?

Reply:
Sorry I think I didn't write it clearer: if there's no text, render null.

Reply:
ahhh that makes sense, thanks!